articles/flask-crud-api/index.md

```diff
@@ -0,0 +1,306 @@
+In this tutorial, we will create a fully working Restful API with flask and Mysql database. Restful APIs makes it possible to establish communication between the backend and the frontend mobile and web applications, we will be using Flask-SQLAlchemy. Flask-SQLAlchemy is an extension python library for the flask web framework that adds SQLAlchemy to the flask project. SQLAlchemy is an Object Relational Mapper that provides access to SQL databases through python objects.
+
+### Prerequisites
+1. [Python](https://www.python.org/downloads/) installed on your computer.
+2. [Postman](https://www.postman.com/downloads/) installed on your computer.
+3. [Apache XAMPP](https://www.apachefriends.org/download.html) installed on your computer.
+4. Favourite IDE installed. I use [Pycharm](https://www.jetbrains.com/pycharm/download/).
+5. Basic [Python](https://www.w3schools.com/python/) knowledge. 
+
+### Project setup
+On the terminal execute the below command to create the projects' working directory and move into that directory.
+```bash
+$ mkdir flasktodo
+cd flasktodo
+```
+In the projects' working directory execute the below command to create a virtual environment for our project. Virtual environments make it easier to manage packages for various projects separately.
+```bash
+$ virtualenv venv
```

Comment:
Virtualenv must be installed on your machine before running this command. Consider adding it to the prerequisites.